docs/Secure-Coding-Guide-for-Python/CWE-664/CWE-197/01/README.md

```diff
@@ -0,0 +1,105 @@
+# CWE-197: Control rounding when converting to less precise numbers
+
+While defensive coding requires enforcing types, it is important to make conscious design decisions on how conversions are rounded.
+
+The `example01.py` code demonstrates how `int()` behaves differently to `round()`.
+
+[*example01.py:*](example01.py)
+
+```py
+""" Code Example """
+
+print(int(0.5))   # prints 0
+print(int(1.5))   # prints 1
+print(int(1.45))  # prints 1
+print(int(1.51))  # prints 1
+print(int(-1.5))  # prints -1
+
+print(round(0.5))   # prints 0
+print(round(1.5))   # prints 2
+print(round(1.45))  # prints 1
+print(round(1.51))  # prints 2
+print(round(-1.5))  # prints -2
+
+print(type(round(0.5)))  # prints <class 'int'>
+
+```
+
+The build in `round()` does not allow to specify the type of rounding in use [[python round( ) 2024]](https://docs.python.org/3/library/functions.html#round). In Python 3 the `round()` function uses "bankers' rounding" (rounds to the nearest even number in case of ties). This is different to Python 2 which always rounds away from zero. Rounding provided by the decimal module allows a choice between 8 rounding modes [python decimal 2024](https://docs.python.org/3/library/decimal.html#rounding-modes). Rounding in mathematics and science is not discussed here as it requires a deeper knowledge of computer floating-point arithmetic's.
```

Comment:
cosmetics: 
- Rounding provided by the `decimal` module 
- Square brackets around sources: 
-- [[python round( ) 2024](https://docs.python.org/3/library/functions.html#round)]
-- [[python decimal 2024](https://docs.python.org/3/library/decimal.html#rounding-modes)]